WordPress/Jetpack/Classes/Utility/DataMigrator.swift

```
@@ -0,0 +1,129 @@
+final class DataMigrator {
+
+    private let coreDataStack: CoreDataStack
+    private let backupLocation: URL?
+    private let keychainUtils: KeychainUtils
+    private let localDefaults: UserDefaults
+    private let sharedDefaults: UserDefaults?
+
+    init(coreDataStack: CoreDataStack = ContextManager.sharedInstance(),
+         backupLocation: URL? = FileManager.default.containerURL(forSecurityApplicationGroupIdentifier: "group.org.wordpress")?.appendingPathComponent("WordPress.sqlite"),
+         keychainUtils: KeychainUtils = KeychainUtils(),
+         localDefaults: UserDefaults = UserDefaults.standard,
+         sharedDefaults: UserDefaults? = UserDefaults(suiteName: WPAppGroupName)) {
+        self.coreDataStack = coreDataStack
+        self.backupLocation = backupLocation
+        self.keychainUtils = keychainUtils
+        self.localDefaults = localDefaults
+        self.sharedDefaults = sharedDefaults
+    }
+
+    enum DataMigratorError: Error {
+        case localDraftsNotSynced
+        case databaseCopyError
+        case keychainError
+        case sharedUserDefaultsNil
+    }
+
+    func exportData(completion: ((Result<Void, DataMigratorError>) -> Void)? = nil) {
+        guard isLocalDraftsSynced() else {
+            completion?(.failure(.localDraftsNotSynced))
+            return
+        }
```

So if the user has any local drafts not sync'ed to remote, the export process will fail early. Am I correct? In that case, can we perhaps try to forcefully sync those drafts before proceeding with the export process?

Or did you intend to offload the responsibility to the caller? (If so, perhaps it would be helpful to publicize the `isLocalDraftsSynced()` method?)

---

Yeah, we need to force sync the local drafts before we export the data. This is still one of the tasks that need to be completed.

We could make that function public if it made it easier for the caller. I'll leave it up to whoever picks up the local draft syncing task for what they want to do.